### PR TITLE
fix(table): differentiate interactivity of rows vs. components in cells

### PR DESCRIPTION
**Pull request by Kiarokh**
related to https://github.com/Lundalogik/crm-feature/issues/2757

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/table/table.tsx [hunk 1]
```diff
@@ -683,6 +683,11 @@ export class Table {
         } else {
             row.getElement().classList.remove('active');
         }
+
+        const element = row.getElement().ownerDocument.createElement('div');
+        element.classList.add('interactive-feedback');
+
+        row.getElement().prepend(element);
```
Comment (Kiarokh) [inline src/components/table/table.tsx:690]: the problem with this implementation is that when the table loads, you get one such a div inside each row. 

But every time you click on a row, a new div with this class will be added to that row again
image

File: src/components/table/table.scss [hunk 2]
```diff
@@ -130,6 +130,10 @@ $table--limel-table--row-selector: 1;
             justify-content: center;
         }
     }
+
+    .tabulator-col-resize-handle {
+        display: none;
+    }
```
Comment (Kiarokh) [inline src/components/table/table.scss:136]: removes this annoying effect:
https://github.com/Lundalogik/lime-elements/assets/35954987/16317cbb-c59d-4653-a954-0c10c7526f96

File: src/components/table/partial-styles/tabulator-custom-styles.scss [hunk 3]
```diff
@@ -1,32 +1,87 @@
...
+    // We use it to visualize interactive feedback.
+    position: absolute;
+    inset: 0;
+    min-height: 0.125rem;
+
+    // We could use `@include mixins.is-flat-clickable()`,
+    // But the "arguments" of the mixin would make the CSS for
+    // even and odd row colors and background color very complicated.
+    // So instead, I've copy/pasted the useful parts of the mixin here
+    transition: color 0.2s ease, background-color 0.2s ease,
```
Comment (LawrenceBorst) [inline src/components/table/partial-styles/tabulator-custom-styles.scss:15]: But do we need transform here? It never really transforms (as in scales/rotates etc.)?

File: src/components/table/partial-styles/tabulator-custom-styles.scss [hunk 4]
```diff
@@ -1,32 +1,87 @@
...
+.tabulator-row:not(.tabulator-calcs) {
+    .tabulator-cell {
+        // We don't want the cells to block
+        // the hovering possibility of `interactive-feedback` div.
+        pointer-events: none;
+        // but we want the content in them
+        // (e.g. links, phone numbers, chips, components)
+        // to remain interactive.
+        > * {
+            pointer-events: all;
```
Comment (LawrenceBorst) [inline src/components/table/partial-styles/tabulator-custom-styles.scss:28]: Should we use auto here?

File: src/components/table/partial-styles/tabulator-custom-styles.scss [hunk 5]
```diff
@@ -1,32 +1,87 @@
...
+            }
+
+            &:before {
+                $width-of-selected-row-indicator: 0.2rem;
+                content: '';
+                display: inline-block;
+                box-sizing: border-box;
+                position: sticky;
+                z-index: $table--has-interactive-rows--selectable-row--hover + 1;
+                inset: 0 0 auto 0.25rem;
```
Comment (LawrenceBorst) [inline src/components/table/partial-styles/tabulator-custom-styles.scss:52]: But the inset used to be 0 0 auto 0, just curious why it changed 😅

File: src/components/table/partial-styles/tabulator-custom-styles.scss [hunk 6]
```diff
@@ -1,32 +1,87 @@
...
-
             &:not(.active) {
-                &:hover {
-                    background-color: rgb(
-                        var(--table-row-background-color--hover)
-                    );
+                .interactive-feedback {
+                    &:hover {
+                        background-color: var(
+                            --table-row-background-color--hover
```
Comment (LawrenceBorst) [inline src/components/table/partial-styles/tabulator-custom-styles.scss:82]: Nice to remove those unnecessary "rgb"'s :)
There's a few left, but this doesn't really matter

**Review by LawrenceBorst — COMMENTED** [review]
Overall looks good
I'll make a fixup for the tests and maybe we can huddle

File: src/components/table/partial-styles/tabulator-custom-styles.scss (hunk 4 above)
Comment (Kiarokh) [reply src/components/table/partial-styles/tabulator-custom-styles.scss:28]: oh yes. good point!

File: src/components/table/partial-styles/tabulator-custom-styles.scss (hunk 5 above)
Comment (Kiarokh) [reply src/components/table/partial-styles/tabulator-custom-styles.scss:52]: Oh I was hoping the reviewer will miss this 😄 

I was too lazy and didn't really want to make a new commit for this change.

I just felt that the green circle was too close to the edge.
image
But I'll take it out from this commit now.

File: src/components/table/partial-styles/tabulator-custom-styles.scss (hunk 4 above)
Comment (Kiarokh) [reply src/components/table/partial-styles/tabulator-custom-styles.scss]: https://github.com/Lundalogik/lime-elements/assets/35954987/28ab243e-7e57-4c4f-be88-6a4b0208eaf5

**Review by LawrenceBorst — APPROVED** [review]

File: src/components/table/partial-styles/tabulator-custom-styles.scss (hunk 3 above)
Comment (Kiarokh) [reply src/components/table/partial-styles/tabulator-custom-styles.scss:15]: when the row is pressed (`active`), it slightly moves down. that's the transform

**Review by LawrenceBorst — APPROVED** [review]

**Pull request merged**